pandas/tests/indexes/test_interval.py

```diff
@@ -327,171 +335,297 @@ def test_get_item(self):
+                    (Int64Index([2], dtype='int64'), array([]))
+                    (Int64Index([2], dtype='int64'), array([]))
+                    (Int64Index([], dtype='int64'), array([0]))
+                    (Int64Index([], dtype='int64'), array([0]))]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer_non_unique([query])
+            tm.assert_numpy_array_equal(result, expected_result)
+
+        # multiple queries
+        queries = [[1, 2], [1, 2, 3], [1, 2, 3, 4], [1, 2, 3, 4, 2]]
+        expected = [(Int64Index([0, 1, 0, 1, 2], dtype='int64'), np.array([]))
+                    (Int64Index([0, 1, 0, 1, 2, 2],
+                                dtype='int64'), np.array([]))
+                    (Int64Index([0, 1, 0, 1, 2, 2, -1],
+                                dtype='int64'), np.array([3]))
+                    (Int64Index([0, 1, 0, 1, 2, 2, -1, 0, 1, 2],
+                                dtype='int64'), np.array([3]))]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer_non_unique(query)
+            tm.assert_numpy_array_equal(result, expected_result)
+
+        # we may also want to test get_indexer for the case when
+        # the intervals are duplicated, decreasing, non-monotonic, etc..
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_contains_updated_behavior(self):
+
+        index = IntervalIndex.from_arrays([0, 1], [1, 2], closed='right')
+
+        # __contains__ requires perfect matches to intervals.
         assert 0 not in i
         assert 1 not in i
         assert 2 not in i
 
-        # Valid
-        assert Interval(0, 1) in i
-        assert Interval(0, 2) in i
-        assert Interval(0, 0.5) in i
-        assert Interval(3, 5) not in i
-        assert Interval(-1, 0, closed='left') not in i
+        assert Interval(0, 1, closed='right') in index
+        assert Interval(0, 2, closed='right') not in index
+        assert Interval(0, 0.5, closed='right') not in index
+        assert Interval(3, 5, closed='right') not in index
+        assert Interval(-1, 0, closed='left') not in index
+        assert Interval(0, 1, closed='left') not in index
+        assert Interval(0, 1, closed='both') not in index
 
-    def testcontains(self):
-        # can select values that are IN the range of a value
-        i = IntervalIndex.from_arrays([0, 1], [1, 2])
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def testcontains_updated_behavior(self):
 
-        assert i.contains(0.1)
-        assert i.contains(0.5)
-        assert i.contains(1)
-        assert i.contains(Interval(0, 1))
-        assert i.contains(Interval(0, 2))
+        index = IntervalIndex.from_arrays([0, 1], [1, 2], closed='right')
 
-        # these overlaps completely
-        assert i.contains(Interval(0, 3))
-        assert i.contains(Interval(1, 3))
+        assert not index.contains(0)
```

Comment:
I am not sure whether we discussed this in this PR, but I think `contains` should return a boolean mask instead of a single boolean value (`__contains__` already does that)